tests/unit/writers/test_qbwriter.py

```diff
@@ -745,30 +764,424 @@ def test_csv_col_definition_suppressed():
 
-    virt_unit = first(virtual_columns, lambda x: x["name"] == "virt_unit")
-    assert virt_unit is not None
-    assert virt_unit["virtual"]
+    cube = Cube(
+        CatalogMetadata("Cube"),
+        columns=[
+            QbColumn("Some Dimension", NewQbDimension("Some Dimension")),
+            QbColumn("Some Attribute", NewQbAttribute(label="Some Attribute")),
+            expected_obs_val_col,
+        ],
+    )
+
+    writer = QbWriter(cube)
+    actual_obs_val_col = writer._get_observation_value_col_for_column("Some Obs Val")
+
+    assert actual_obs_val_col == expected_obs_val_col
+
+
+def test_get_observation_value_col_for_column_when_col_title_is_invalid():
+    """
+    Ensure that the invalid column title raises an exception.
+    """
+
+    cube = Cube(
+        CatalogMetadata("Cube"),
+        columns=[
+            QbColumn("Invalid Col Title", NewQbDimension("Some Dimension")),
+            QbColumn("Some Attribute", NewQbAttribute(label="Some Attribute")),
+            QbColumn(
+                "Some Obs Val",
+                QbObservationValue(
+                    NewQbMeasure("Some Measure"), NewQbUnit("Some Unit")
+                ),
+            ),
+        ],
+    )
+
+    writer = QbWriter(cube)
+    with pytest.raises(Exception) as err:
+        writer._get_observation_value_col_for_column("Invalid Col Title")
+
     assert (
-        "http://purl.org/linked-data/sdmx/2009/attribute#unitMeasure"
-        == virt_unit["propertyUrl"]
+        str(err.value)
+        == 'Could not find one observation value column. Found 0 for title: "Invalid Col Title".'
+    )
+
+
+def test_get_cross_measures_slice_key_for_new_dimension():
+    """
+    Ensure that given a cube with NewQbDimension, the function returns the correct slice key.
+    """
+    cube = Cube(
+        CatalogMetadata("Cube"),
+        columns=[
+            QbColumn("Some Dimension", NewQbDimension("Some Dimension")),
+            QbColumn("Some Attribute", NewQbAttribute(label="Some Attribute")),
+            QbColumn(
+                "Some Obs Val",
+                QbObservationValue(
+                    NewQbMeasure("Some Measure"), NewQbUnit("Some Unit")
+                ),
+            ),
+        ],
+    )
+    writer = QbWriter(cube)
+
+    actual_slice_key = writer._get_cross_measures_slice_key()
+    assert str(actual_slice_key.uri) == "cube.csv#slice/cross-measures"
+
+    component_properties = list(actual_slice_key.componentProperties)
+    assert len(component_properties) == 1
+    assert str(component_properties[0].uri) == "cube.csv#dimension/some-dimension"
+
+
+def test_get_cross_measures_slice_key_for_existing_dimension():
+    """
+    Ensure that given a cube with ExistingQbDimension, the function returns the correct slice key.
+    """
+    cube = Cube(
+        CatalogMetadata("Cube"),
+        columns=[
+            QbColumn("Some Dimension", ExistingQbDimension("Some Dimension")),
```

Comment:
You should be providing a URI to `ExistingQbDimension` for it to be valid.

Can you make sure that everywhere you're using `ExistingQbDimension`, you make sure to pass some kind of URI in to it? I know this problem tripped up @NickPapONS yesterday since he was using these configurations as a basis for another test.

Reply:
Good spot - we have updated.